### PR TITLE
Add permissions to create

### LINKED ISSUE #160: Create page does not have permission dialog
We need it for iOS room creators, or else they'll fat-kid themselves 😢

### PR DESCRIPTION
**Pull request by juxd**
Fix #160

### CODE REVIEW

File: Web/src/components/create-room/CreateRoomPage.tsx [hunk 1]
```diff
@@ -35,18 +39,43 @@ const CreateRoomContainer = styled.div`
   }
 `;
 
+const ErrorText = styled.p`
+  color: red;
```
Comment (hzxcaryn) [inline Web/src/components/create-room/CreateRoomPage.tsx]:
```suggestion
  color: var(--red);
```

File: Web/src/components/room/PermissionsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Modal from 'components/common/Modal';
+
+interface PermissionsDialogType {
```
Comment (hzxcaryn) [inline Web/src/components/room/PermissionsDialog.tsx]: should this be props?
```suggestion
interface PermissionsDialogProps {
```

File: Web/src/components/room/hooks/permission/useMotionPermissionsAccess.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  );
+
+  return {
+    /** The current state of the permissions for the device motion. */
+    permission,
+    /** STATE TO OBSERVE to activate/deactivate request modal. */
+    awaitingPermission,
+    /** Method to call to check/request for permissions on user's phone. */
+    getUserPermission,
+    /** Method to check fro permissions on user's phone, and */
```
Comment (hzxcaryn) [inline Web/src/components/room/hooks/permission/useMotionPermissionsAccess.ts]:
```suggestion
    /** Method to check for permissions on user's phone */
```

File: Web/src/components/create-room/CreateRoomPage.tsx (hunk 1 above)
Comment (juxd) [reply Web/src/components/create-room/CreateRoomPage.tsx]: should we rename this to a more general 'error' color?

**Comment by juxd** [comment]
oh wait no i'm still testing haha

**Pull request merged**